internal/statemachine/classic.go

```diff
@@ -92,12 +92,20 @@ func (stateMachine *StateMachine) parseImageDefinition() error {
 		return err
 	}
 
+	if imageDefinition.Rootfs.SourcesListDeb822 == nil {
+		fmt.Print("WARNING: rootfs.sources-list-deb822 was not set. Please explicitely set the format desired for sources list in your image definition.\n")
+	}
+
 	// populate the default values for imageDefinition if they were not provided in
 	// the image definition YAML file
 	if err := helperSetDefaults(&imageDefinition); err != nil {
 		return err
 	}
 
+	if !*imageDefinition.Rootfs.SourcesListDeb822 {
+		fmt.Print("WARNING: rootfs.sources-list-deb822 is set to false. The deprecated format will be used to manage sources list. Please if possible adopt the new format.\n")
```

Comment:
I'm good with this so far, I think this is okay. But before we release this officially in ubuntu-image, I'd like to understand a bit better if we *actually* can use the old sources.list format (non-deb822) in noble successfully. I know the migration to deb822 in noble is going full force. Now the question is - once the migration is complete, will it still be possible to build a non-deb822 24.04 Ubuntu image and have everything working correctly? Or is the expectation that only deb822 is what we support?

I'm asking this question since maybe, if a non-deb822 would not be functional for noble, we should just either default to 'true' for sources-list-deb822 for all noble+ images or error out when the value is different than expected? Do you know?

Reply:
non-deb822 still should be possible. I suggested defaulting but the goal was to keep release-specific knowledge out.

The warning I suggested to make it clear that the default will change eventually but arguably you don't need it *now*. Give people 6 months or so to update before adding a warning.

Also the warning needs to be a bit more careful I suppose, we don't want to suggest people to build pre-noble images with deb822 really.

Reply:
Warning to have today, with sources-list-deb822 to false:

`WARNING: rootfs.sources-list-deb822 is set to false. The deprecated format will be used to manage sources list. Please if possible adopt the new format. `

Warning to have today, with sources-list-deb822 to true:

`WARNING: rootfs.sources-list-deb822 is set to true. The DEB822 format will be used to manage sources list. Please make sure you are not building an image older than noble.`

What do you think?

There is already a warning if this value was not specified in the image definition. I am not sure what warning you would like to see in 6 months.